File: MADVCamera/CameraManager/ImageFilterBean.m
//
//  ImageFilterBean.m
//  Madv360_v1
//
//  Created by 张巧隔 on 16/8/10.
//  Copyright © 2016年 Cyllenge. All rights reserved.
//

#import "ImageFilterBean.h"
#ifdef MADVPANO_BY_SOURCE
#import "GLFilterCache.h"
#else
#import <MADVPano/GLFilterCache.h>
#endif

@interface ImageFilterBean ()
@end

@implementation ImageFilterBean

- (instancetype) initWithUUID:(int)uuid name:(NSString *)name enName:(NSString *)enName
                  iconPNGPath:(NSString *)iconPNGPath roundIconPNGPath:(NSString*)roundIconPNGPath {
    if (self = [super init])
    {
        self.uuid = uuid;
        self.name = name;
        self.enName = enName;
        self.iconPNGPath = iconPNGPath;
        self.roundIconPNGPath = roundIconPNGPath;
    }
    return self;
}

+ (NSMutableArray *)allImageFilters
{
    static dispatch_once_t once;
    static NSMutableArray* imageFilters;
    dispatch_once(&once, ^{
        typedef struct {
            int filterID;
            NSString* zhName;
            NSString* enName;
            NSString* iconPathName;
            NSString* roundIconPathName;
        } ImageFilterItem;
        static ImageFilterItem items[] = {
            {GLFilterNone, ORIGINAL, @"Original", @"nothing_1", @"nothing_2"},
//            {GLFilterBilateralID, BILATERAL, nil), @"Bilateral", @"exfoliating_1", @"exfoliating_2"},
            {GLFilterSepiaToneID, SEPIATONE, @"Sepia Tone", @"past_1", @"past_2"},
            {GLFilterAmatorkaID, AMATORKA, @"Amatorka", @"young_1", @"young_2"},
//            //{GLFilterMissEtikateID, MISSETIKATE, @"Miss Etikate", @"sea_1", @"sea_2"},
//            {GLFilterInverseColorID, FILM, @"Color Inverse", @"film_1", @"film_2"},
            {GLFilterToonID, TOON, @"Toon", @"toon_1", @"toon_2"},
            {GLFilterSobelEdgeDetectSketchID, SKETCH, @"Sketch", @"sketch_1", @"sketch_2"},
        };
        imageFilters = [[NSMutableArray alloc] init];
        NSBundle* bundle = [NSBundle mainBundle];
        for (int i = 0; i < sizeof(items) / sizeof(ImageFilterItem); ++i)
        {
            ImageFilterItem item = items[i];
            NSString* iconPNGPath = [bundle pathForResource:item.iconPathName ofType:@"png"];
            NSString* roundIconPNGPath = [bundle pathForResource:item.roundIconPathName ofType:@"png"];
            ImageFilterBean* bean = [[ImageFilterBean alloc] initWithUUID:item.filterID name:item.zhName enName:item.enName iconPNGPath:iconPNGPath roundIconPNGPath:roundIconPNGPath];
            [imageFilters addObject:bean];
        }
    });
    return imageFilters;
}

+ (ImageFilterBean *)findImageFilterByID:(int)uuid {
    NSMutableArray* imageFilters = [self allImageFilters];
    for (ImageFilterBean* filter in imageFilters)
    {
        if (filter.uuid == uuid)
            return filter;
    }
    return nil;
}

@end

File: MADVCamera/CameraManager/ImageFilterBean.java
package com.madv360.madv.model.bean;

import com.madv360.glrenderer.GLFilterCache;
import com.madv360.madv.R;

import java.util.ArrayList;

/**
 * Created by qiudong on 16/6/17.
 */
public class ImageFilterBean {

    public int uuid;

    public String name;

    public String enName;

    public String iconPNGUri1;

    public String iconPNGUri2;

    public ImageFilterBean() {

    }

    public ImageFilterBean(int uuid, String name, String enName, String uri1, String uri2) {
        this.uuid = uuid;
        this.name = name;
        this.enName = enName;
        this.iconPNGUri1 = uri1;
        this.iconPNGUri2 = uri2;
    }

    private static ArrayList<ImageFilterBean> ImageFilters = null;

    public static final synchronized ArrayList<ImageFilterBean> allImageFilters() {
        if (null == ImageFilters) {
            ImageFilters = new ArrayList<>();
            ImageFilters.add(new ImageFilterBean(GLFilterCache.GLFilterNone, "原图", "Original"
                    , "res://com.madv360.madv/" + R.drawable.nothing_1
                    , "res://com.madv360.madv/" + R.drawable.nothing_2));
            ImageFilters.add(new ImageFilterBean(GLFilterCache.GLFilterBilateralID, "磨皮", "Bilateral"
                    , "res://com.madv360.madv/" + R.drawable.exfoliating_1
                    , "res://com.madv360.madv/" + R.drawable.exfoliating_2));
            ImageFilters.add(new ImageFilterBean(GLFilterCache.GLFilterSepiaToneID, "流年", "Sepia Tone"
                    , "res://com.madv360.madv/" + R.drawable.past_1
                    , "res://com.madv360.madv/" + R.drawable.past_2));
            ImageFilters.add(new ImageFilterBean(GLFilterCache.GLFilterAmatorkaID, "青春", "Amatorka"
                    , "res://com.madv360.madv/" + R.drawable.young_1
                    , "res://com.madv360.madv/" + R.drawable.young_2));
            ImageFilters.add(new ImageFilterBean(GLFilterCache.GLFilterMissEtikateID, "海韵", "Miss Etikate"
                    , "res://com.madv360.madv/" + R.drawable.sea_1
                    , "res://com.madv360.madv/" + R.drawable.sea_2));
            ImageFilters.add(new ImageFilterBean(GLFilterCache.GLFilterInverseColorID, "胶片", "Color Inverse"
                    , "res://com.madv360.madv/" + R.drawable.film_1
                    , "res://com.madv360.madv/" + R.drawable.film_2));
        }
        return ImageFilters;
    }

    public static final ImageFilterBean findImageFilterByID(int uuid) {
        ArrayList<ImageFilterBean> filters = allImageFilters();
        for (ImageFilterBean filter : filters) {
            if (filter.uuid == uuid) {
                return filter;
            }
        }
        return null;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof ImageFilterBean))
            return false;

        ImageFilterBean other = (ImageFilterBean) o;
        return (other.uuid == this.uuid);
    }

    @Override
    public int hashCode() {
        return uuid;
    }
}
